File: puntos_generados.py
import tkinter as tk
import random

#dimencion de la ventana
WINDOW_WIDTH = 600
WINDOW_HEIGHT = 400

#Cantidad de puntos
NUM_PUNTOS = 100
#Radio de los puntos
POINT_RADIUS = 5

#Funcion para generar un punto aleatorio
def generate_random_point():
    x = random.randint(POINT_RADIUS, WINDOW_WIDTH - POINT_RADIUS)
    y = random.randint(POINT_RADIUS, WINDOW_HEIGHT - POINT_RADIUS)
    return (x, y)

#Funcion para verificar si dos puntos se solapan
def points_overlap(p1, p2):
    distance = ((p1[0] - p2[0]) ** 2 + (p1[1] - p2[1]) ** 2 + (p1[1] - p2[1]) ** 2) ** 0.5
    return distance < 2 * POINT_RADIUS

#Algoritmo de las vegas para generar puntos sin solapamientos
def generate_pionts(num_points):
    points = []
    while len(points) < num_points:
        point = generate_random_point()
        if all(not points_overlap(point, existing_point) for existing_point in points):
            points.append(point)
    return points

#Funcion para dibujar los puntos en el lienzo
def draw_points(canvas, points):
    for (x, y) in points:
        canvas.create_oval(x - POINT_RADIUS, y - POINT_RADIUS, x + POINT_RADIUS, y + POINT_RADIUS, fill="blue")
        
#Crear la ventana principal
root = tk.Tk()
root.title("Puntos aleatorios con Algoritmo Las Vegas")

#Crear el lienzo
canvas = tk.Canvas(root, width=WINDOW_WIDTH, height=WINDOW_HEIGHT, bg="white")
canvas.pack()

#Generar y dibujar los puntos
points = generate_pionts(NUM_PUNTOS)
draw_points(canvas, points)

#Iniciar el bucle principal de Tkinter
root.mainloop()

File: puntos_generados.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#include <math.h>

#define WINDOW_WIDTH 600
#define WINDOW_HEIGHT 400
#define NUM_POINTS 50
#define POINT_RADIUS 5

// Estructura para un punto
typedef struct {
    int x;
    int y;
} Point;

// Función para generar un punto aleatorio
Point generate_random_point() {
    Point p;
    p.x = POINT_RADIUS + rand() % (WINDOW_WIDTH - 2 * POINT_RADIUS);
    p.y = POINT_RADIUS + rand() % (WINDOW_HEIGHT - 2 * POINT_RADIUS);
    return p;
}

// Función para verificar si dos puntos se solapan
int points_overlap(Point p1, Point p2) {
    double distance = sqrt((p1.x - p2.x) * (p1.x - p2.x) + (p1.y - p2.y) * (p1.y - p2.y));
    return distance < 2 * POINT_RADIUS;
}

// Algoritmo de Las Vegas para generar puntos sin solapamientos
void generate_points(Point *points, int num_points) {
    int count = 0;
    while (count < num_points) {
        Point point = generate_random_point();
        int overlap = 0;
        for (int i = 0; i < count; i++) {
            if (points_overlap(point, points[i])) {
                overlap = 1;
                break;
            }
        }
        if (!overlap) {
            points[count] = point;
            count++;
        }
    }
}

int main() {
    srand(time(NULL));  // Inicializar la semilla para números aleatorios

    Point points[NUM_POINTS];
    generate_points(points, NUM_POINTS);

    // Imprimir los puntos generados
    printf("Puntos generados:\n");
    for (int i = 0; i < NUM_POINTS; i++) {
        printf("Punto %d: (%d, %d)\n", i + 1, points[i].x, points[i].y);
    }

    return 0;
}
